components/tidb_query/src/rpn_expr/impl_encryption.rs

```diff
@@ -26,6 +32,26 @@ pub fn sha1(arg: &Option<Bytes>) -> Result<Option<Bytes>> {
     }
 }
 
+#[rpn_fn]
+#[inline]
+pub fn sha2(input: &Option<Bytes>, hash_length: &Option<Int>) -> Result<Option<Bytes>> {
+    match (input, hash_length) {
+        (Some(input), Some(hash_length)) => {
+            let sha2 = match *hash_length {
+                SHA0 | SHA256 => MessageDigest::sha256(),
+                SHA224 => MessageDigest::sha224(),
+                SHA384 => MessageDigest::sha384(),
+                SHA512 => MessageDigest::sha512(),
+                _ => {
+                    return Err(Error::incorrect_parameters("sha2").into());
```

Comment:
There should be a warning in context and return `Ok(None)` when  `hash_length`  is invalid.

Reply:
Could you please explain why we can not return err?
I think the parameters incorrect should return err.

Reply:
We need the same behavior as MySQL does.

https://dev.mysql.com/doc/refman/5.5/en/encryption-functions.html#function_sha2

>  If either argument is NULL or the hash length is not one of the permitted values, the return value is NULL

and you can check the behavior in MySQL console
```
mysql> select sha2("abc", 12);
+-----------------+
| sha2("abc", 12) |
+-----------------+
| NULL            |
+-----------------+
1 row in set, 1 warning (0.01 sec)

mysql> show warnings;
+---------+------+------------------------------------------------------------+
| Level   | Code | Message                                                    |
+---------+------+------------------------------------------------------------+
| Warning | 1583 | Incorrect parameters in the call to native function 'sha2' |
+---------+------+------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
Thank  you, got it. 
I will add it tonight.

Reply:
modified